File: 1480. Running Sum of 1d Array.java
/**
 * https://leetcode.com/problems/running-sum-of-1d-array/
 */

public class RunningSumOf1dArray {
    // O(n) time complexity
    // O(1) space complexity
    int[] runningSum(int[] nums) {
        for (int i = 1; i < nums.length; i++)
            nums[i] += nums[i - 1];
        return nums;
    }
}

class T {
    public static void main(String[] args) {
        RunningSumOf1dArray r = new RunningSumOf1dArray();
        int[] nums = {1, 2, 3, 4};
        r.runningSum(nums);
        for (int num : nums)
            System.out.print(num + " "); // output: 1, 3, 6, 10
    }
}

File: Kotlin/1480. Running Sum of 1d Array.kt
/**
 * https://leetcode.com/problems/running-sum-of-1d-array/
 */
// O(n) time complexity
// O(1) space complexity
fun runningSum(nums: IntArray): IntArray {
    for (i in 1 until nums.size) {
        nums[i] += nums[i - 1]
    }
    return nums
}

fun main() {
    val n = intArrayOf(1, 2, 3, 4)
    runningSum(n)
    n.forEach {
        print("$it ")
    }
}